onnx_tf/handlers/backend/pad.py

```diff
@@ -15,32 +15,66 @@ class Pad(BackendHandler):
   @classmethod
   def _common(cls, node, **kwargs):
     tensor_dict = kwargs["tensor_dict"]
-    attrs = copy.deepcopy(node.attrs)
-    pads = attrs.pop("pads", attrs.pop("paddings", None))
+    x = tensor_dict[node.inputs[0]]
     num_dim = len(tensor_dict[node.inputs[0]].get_shape())
-    mode = attrs.get("mode", "constant")
-    attrs["constant_values"] = attrs.pop("value", 0.)
 
-    def _compatibility_edge_pad(x, pads):
-      x = np.pad(x, pads, mode="edge")
-      return x
+    def _symmetric_pad(i, x):
+      pads_i = tf.map_fn(lambda e: tf.where(i < e, 1, 0), pads)
+      pads_i = tf.reshape(pads_i, [num_dim, 2])
+      x = tf.pad(x, pads_i, 'SYMMETRIC')
+      return i + 1, x
 
-    # tf requires int32 paddings
-    attrs["paddings"] = tf.constant(
-        np.transpose(np.array(pads).reshape([2, num_dim]).astype(np.int32)))
+    if len(node.inputs) == 1:  # for opset 1 and opset 2
+      attrs = copy.deepcopy(node.attrs)
+      pads = attrs.pop("pads", attrs.pop("paddings", None))
+      mode = attrs.get("mode", "constant")
+      attrs["constant_values"] = attrs.pop("value", 0.)
+
+      # tf requires int32 paddings
+      attrs["paddings"] = tf.constant(
+          np.transpose(np.array(pads).reshape([2, num_dim]).astype(np.int32)))
+
+      if mode.lower() == "edge":
+        pads = tf.reshape(attrs["paddings"], [-1])
+        max_i = tf.reduce_max(pads)
+        _, x = tf.while_loop(
+            lambda i, x: tf.less(i, max_i), _symmetric_pad, [0, x],
+            [tf.TensorShape([]), tf.TensorShape(None)])
+        return [x]
```

Comment:
Seems same block of code for opset 11 except pads coming from attribute. Consider using a common method.